src/dune_pkg/lock_dir.ml

```diff
@@ -9,6 +9,27 @@ module Source = struct
         ; checksum : (Loc.t * Checksum.t) option
         }
 
+  let equal a b =
```

Comment:
Maybe this should be called equal_no_locs or something since we are explicitly ignoring the loc. Also what is the loc from/used for?

Reply:
You could also just add a `drop_loc` function to convert all the locations into `Loc.none`. This is the convention we use elsewhere.

Reply:
@Alizter the loc is used in error messages. Will update to have a `drop_loc` function.